#!/usr/bin/env python

PACKAGE='c2_hw'

import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()
#        Name            Type      Level  Description                           Default   Min       Max
gen.add( "p" ,           double_t, 1,     "Proportional gain.",                 10.0,     -100000,  100000)
gen.add( "i" ,           double_t, 1,     "Integral gain.",                     0.1,      -1000,    1000)
gen.add( "d" ,           double_t, 1,     "Derivative gain.",                   1.0,      -1000,    1000)
gen.add( "i_clamp_min" , double_t, 1,     "Min bounds for the integral windup", -10.0,    -1000,    0)
gen.add( "i_clamp_max" , double_t, 1,     "Max bounds for the integral windup", 10.0,     0,        1000)
gen.add( "antiwindup" ,  bool_t,   1,     "Antiwindup.",                        False)

# PkgName  #NodeName         #Prefix for generated .h include file, e.g. ParametersConfig.py
exit(gen.generate(PACKAGE, "c2_hw_node", "PidParams"))
